### PR TITLE
JetBrains: Use DialogWrapper instead of JBPopup

### LINKED ISSUE #34967: JetBrains: Save window size
## Context

We currently always open the Find on Sourcegraph popup at a 1200x800px size.
The native "Find in Files" popup remembers its last size and position, which is a quite nice UX.

## Problem

It's frustrating to resize the window at each open when I need it at a certain size

## Solution

Save the last size and position somewhere.
It might get tricky when considering multiple screens and screen resolution changes. Test these cases.

### LINKED ISSUE #37322: JetBrains: Popup stays on top of other apps when clicking away
The popup stays on top of other apps on Mac (didn't test it in other OSes) until two clicks made on the other app:

CleanShot 2022-06-16 at 12 07 48@2x

[45-sec Loom demoing the behavior](https://www.loom.com/share/93fab13bad8645e68843fbacd4416d85)

This is rather annoying for me as a user, and not the way the Find in Files and other popups work in JetBrains IDEs.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #37322
Closes #34967

This PR rewrites the way we handle the popover inside Java to use the DialogWrapper (the same primitive that is underneath the native Find in Files dialog) instead of a JBPopup.

There are many considerations that went into this with a lot of trial and error but I will try to recap this as good as possible:

- The `DialogWrapper` makes the modal behave like it's own window which means that if another process is brought into the forgeround on macOS, it will not be rendered on top of it (Our main motivation for creating this work, c.f. #37322)
- It does by default also look like a native window (with the app close icons etc.). We have to completely opt-out of this behavior via `setUndercorated(true`).
- Undecorated modals do not have any behavior to them. E.g. they won't close if you do anything, they won't have resize listeners, nothing. Thus, we had to re-implement all these behaviors manually. Most of that is inspired by the [native find in files implementation](https://sourcegraph.com/github.com/JetBrains/intellij-community/-/blob/platform/lang-impl/src/com/intellij/find/impl/FindPopupPanel.java) and was moved to `addNativeFindInFilesBehaviors()`.
  - This method is applying all "tweaks" that the native find in files component is also doing. It also has special code to handle restoring of window dimensions that are saved under special considerations (e.g. different screen resolutions), so it is probably a good win for us to rely on that logic and not have to find the same mistakes over time. 
- There are two main changes that we are applying:
  - We don't dispose the window if we no longer need it because of the web view disappearance issue. Instead, we only hide it visually. This needed some special handling to persist the window size as the default behavior only triggers when the window is disposed (which it no longer is). 
  -  We want the window to close when another window gets focus. This needs to be scoped per-project as multiple JetBrains project can be open in the same Application. For that, we could reuse the previous `registerOutsideClickListener` logic.

I’m sorry for the mess in the commit history. Prior to coming back to this approach where we subclass the `DialogWrapper` I worked on [an approach similar to the native find in files](https://github.com/sourcegraph/sourcegraph/tree/ps/jetbrains-find-in-files-fork). I've taken the results and learnings from this work and applied it here in one go so you don't see the individual stages of API surgery that I did before, lol. Here's a quick history of the historical changes: https://github.com/sourcegraph/sourcegraph/commits/ps/jetbrains-find-in-files-fork

## Test plan

_A lot_ of manual testing but I need more help! Here are just some of the cases that I tried:

https://drive.google.com/file/d/1elpP1wF4JtEpDZw07qgRo20NItFYP5Wh/view?usp=sharing

## App preview:

- [Web](https://sg-web-ps-jetbrains-dialogwrapper.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-tnsxpzgckv.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/find/FindService.java [hunk 1]
```diff
@@ -56,44 +44,35 @@ synchronized public void showPopup() {
...
-        registerJBCefClientKeyListeners();
-
-        return builder.createPopup();
+    private void createOrShowPopup() {
+        if (popup != null) {
+            if (!popup.isVisible()) {
+                popup.show();
+            }
+        }
+        if (popup == null) {
```
Comment (vdavid) [inline client/jetbrains/src/main/java/com/sourcegraph/find/FindService.java:58]: Just curious: was it a conscious choice to not make this an `else` block?

**Review by vdavid — APPROVED** [review]
Nice! I hope we get away in the long term with using internal APIs at two places. Overall, I think this is the best solution we can have now. Amazing that you could make it work! :)

File: client/jetbrains/src/main/java/com/sourcegraph/find/FindService.java (hunk 1 above)
Comment (philipp-spiess) [reply client/jetbrains/src/main/java/com/sourcegraph/find/FindService.java:58]: Not really I think this is an artifact from copy-pasting too much things around :D Will clean it up 🙂

**Pull request merged**